doc/upgrades.md

```diff
@@ -16,27 +20,39 @@ To enable an upgrade in Hotshot protocol, it is essential to define the base ver
 These are defined in [NodeType implementation](../types/src/v0/mod.rs) for the Types (`SeqTypes` in our case).
 ```rust
 impl NodeType for SeqTypes {
-    type Base = StaticVersion<0, 1>;
-    type Upgrade = StaticVersion<0, 2>;
-    const UPGRADE_HASH: [u8; 32] = [
-        1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
-    ], 
-    ..
+	type Base = StaticVersion<0, 1>;
+	type Upgrade = StaticVersion<0, 2>;
+	const UPGRADE_HASH: [u8; 32] = [
+		1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
+	],
+	..
 }
 ```
 
-These parameters are fetched from the genesis TOML file and set in Hotshot config:
+Hotshot provides two modes for upgrades: view-based and time-based.
+
+View-based parameters allow nodes to vote and propose an upgrade based on the current node's view, while time-based parameters use Unix timestamps for the same purpose.
 
-- **start_voting_view:**  view at which voting for the upgrade proposal starts. In our implementation, this is set to 1 so that voting is enabled as soon as the node is started.
+To simplify configuration, these parameters are fetched from the genesis TOML file and set in the Hotshot config. The TOML file can include either view-based parameters or time-based parameters, but not both.
+Furthermore, The start and stop voting parameters for both time-based and view-based upgrades are optional. Start parameter is set 0 so that voting begins as soon as node is started  while the stop parameter is set to a maximum value so that the nodes keep voting until enough votes are collected.
+
+View based:
+- **start_voting_view:**  view at which voting for the upgrade proposal starts.
 - **stop_voting_view:**  view at which voting for the upgrade proposal stops. To disable an upgrade, set this parameter to 0 or ensure `stop_voting_view` is less than `start_voting_view`.
 - **start_proposing_view:** the earliest view in which the node can propose an upgrade. This should be set to when an upgrade is intended. If the current view > `start_proposing_view`, the node will send out an `UpgradeProposal`.
 - **stop_proposing_view:** view after which the node stops proposing an upgrade. If the upgrade proposal fails and the current view > stop_proposing_view then the upgrade is never proposed again.
 
-The window between `start_proposing_view` and `stop_proposing_view` should provide sufficient time for nodes to continue proposing the upgrade until successful.
+Time based:
+- **start_voting_time:**  UNIX timestamp at which voting for the upgrade proposal starts.
```

Comment:
What's the use case for having distinct `start_voting_time` and `start_proposing_time`? Like why can't they be the same?

Reply:
I don't think this serves a very useful purpose, but it has been provided as an additional setting. This is why we are keeping the voting parameters optional in the TOML file and setting them to 0/u64::MAX by default.

Reply:
Thanks. I think if it's not useful it would be better not to have it, but at least it's documented now. :D